---------------------
PatchSet 10741 
Date: 2006/06/21 12:25:10
Author: hno
Branch: SQUID_2_5
Tag: (none) 
Log:
Bug #1605: Parent not always logged to access.log

The patch for Bug #1504 forgot to account for persistent connections,
causing NONE/- to be logged in the hierarchy field when using a persistent
peer connection.

A workaround is to set "serverside_persistent_connections off"

Members: 
	src/forward.c:1.82.2.17->1.82.2.18 

Index: squid/src/forward.c
===================================================================
RCS file: /cvsroot/squid/squid/src/forward.c,v
retrieving revision 1.82.2.17
retrieving revision 1.82.2.18
diff -u -r1.82.2.17 -r1.82.2.18
--- squid/src/forward.c	10 Mar 2006 22:54:38 -0000	1.82.2.17
+++ squid/src/forward.c	21 Jun 2006 12:25:10 -0000	1.82.2.18
@@ -380,6 +380,12 @@
 	    if (!fs->peer)
 		fwdState->origin_tries++;
 	    comm_add_close_handler(fd, fwdServerClosed, fwdState);
+	    if (fs->peer)
+		hierarchyNote(&fwdState->request->hier, fs->code, fs->peer->host);
+	    else if (Config.onoff.log_ip_on_direct && fs->code == DIRECT)
+		hierarchyNote(&fwdState->request->hier, fs->code, fd_table[fd].ipaddr);
+	    else
+		hierarchyNote(&fwdState->request->hier, fs->code, fwdState->request->host);
 	    fwdConnectDone(fd, COMM_OK, fwdState);
 	    return;
 	} else {
